src/gicv2.rs

```diff
@@ -0,0 +1,294 @@
+// Copyright 2024 The arm-gic Authors.
+// This project is dual-licensed under Apache 2.0 and MIT terms.
+// See LICENSE-APACHE and LICENSE-MIT for details.
+
+//! Driver for the Arm Generic Interrupt Controller version 2.
+
+mod registers;
+
+use self::registers::{GicdCtlr, GICC, GICD};
+use core::{
+    fmt::{self, Debug, Formatter},
+    ptr::{addr_of, addr_of_mut},
```

Comment:
Please use `&raw const` and `&raw mut` instead, we already use them in the GICv3 driver.